python/arcticdb/storage_fixtures/mongo.py

```diff
@@ -21,6 +22,9 @@
 if TYPE_CHECKING:
     from pymongo import MongoClient
 
+logger = logging.getLogger("Mongo Storage Fixture")
+logging.basicConfig(level=logging.DEBUG)
```

Comment:
Do we want this?

Reply:
I think having informative log messages is useful, but I agree having a default call to `basicConfig` to actually enable them will be frustrating for other users of this server fixture.

Therefore I'd suggest to remove the `basicConfig`, and then for debugging the fixture users can import the `log` object and call `basicConfig` themselves if they would like to. This is how [pytest-server-fixtures](https://github.com/man-group/pytest-plugins/blob/master/pytest-server-fixtures/pytest_server_fixtures/mongo.py) does it.

It's very useful for working on the CI as only the locally managed mongo server can be recreated locally, whereas the `CI_MONGO_HOST ` one from the GitHub service would be harder to set up.

Reply:
Makes sense. How about letting the log level be set with an env var so that it's easy to turn it on for CI debugging?

Reply:
That makes sense. I've used `ARCTICDB_mongo_test_fixture_loglevel` as the env var.

I've also made all the fixture logs to `DEBUG` level, and configure the root logger to be `INFO` since that's the default for all our other logs.

I've manually tested that this works:

- With `INFO` just get a few logs coming through from the locally managed mongo instance, but not from the fixture:

```bash
$ ARCTICDB_mongo_test_fixture_loglevel=INFO python -m pytest -vs tests/integration/arcticdb/test_arctic.py | head -n 20
============================= test session starts ==============================
[cut output]
tests/integration/arcticdb/test_arctic.py::test_library_creation_deletion[mongo-EncodingVersion.V1] 2024-02-14T11:34:57.088+0000 I  CONTROL  [main] Automatically disabling TLS 1.0, to force-enable TLS 1.0 specify --sslDisabledProtocols 'none'
2024-02-14T11:34:57.090+0000 W  ASIO     [main] No TransportLayer configured during NetworkInterface startup
[cut output]
```

- With `DEBUG` you can see the test fixture logs:

```bash
$ ARCTICDB_mongo_test_fixture_loglevel=DEBUG python -m pytest -vs tests/integration/arcticdb/test_arctic.py | head -n 20
============================= test session starts ==============================
[ cut output]
tests/integration/arcticdb/test_arctic.py::test_library_creation_deletion[mongo-EncodingVersion.V1] DEBUG:Mongo Storage Fixture:Env var CI_MONGO_HOST not set, so will try localhost.
DEBUG:Mongo Storage Fixture:Could not connect to localhost, so falling back to managed instance.
About to run: ['mongod', '--port', '15099', '--dbpath', '/tmp/tmp10vn8sdrManagedMongoDBServer']
2024-02-14T11:41:24.576+0000 I  CONTROL  [main] Automatically disabling TLS 1.0, to force-enable TLS 1.0 specify --sslDisabledProtocols 'none'
2024-02-14T11:41:24.579+0000 W  ASIO     [main] No TransportLayer configured during NetworkInterface startup
[cut output]
```